homeassistant/components/climate/honeywell.py

```diff
@@ -182,8 +198,8 @@ def update(self):
                     data = val
 
         except StopIteration:
-            _LOGGER.error("Did not receive any temperature data from the "
-                          "evohomeclient API.")
+            _LOGGER.exception("Did not receive any temperature data from the "
```

Comment:
This whole method actually seems very weird. Why would temperatures throw a `StopIteration` error and how will we know that we have already assigned a value?